### ISSUE TITLE
suggestion std::find command script

### ISSUE THREAD

**Issue by sader1992** [rathena/rathena#3064]
* **rAthena Hash**: a16fe436e34c52a981c6d1c0898ac605ae5af89d

* **Client Date**: 2015-11-04

* **Server Mode**: both

* **Description of suggestion**: 
so i was thinking if we can make script command for that in rathena
since most of the scripts need for loop for array comparison
Example:
```
===================================================
setarray .@myarray,2,50,15,17;
if(arraysearch(.@myarray ,5)){
	//false
}

if(arraysearch(.@myarray ,15)){
	//true
}

===================================================

setarray .@myarray$,"test1","test2","testtttt","testttt1";
if(arraysearch(.@myarray$ ,"test")){
	//false
}

if(arraysearch(.@myarray$ ,"testttt1")){
	//true
}
===================================================
```

so if the element is found the command script return true
else return false

(also auto freeloop it if the array is big [i don't know if this is necessary since it's in the src])

i know std::find for vectors but well it's just an idea

`( i did try to make it but no success xD )`

it would take time until all the scripters use it but it would be nice command script

Edit:
thinking about it more
it's better to be like this if we will loop inside the src
```
searcharray(<array>,<"value1">{,<"value2">}[etc up for some number])
return 0 if non is true
return 1 if all true
return 2 if one is true
```
this would be a shortcut for 
```
if(value == value1 || value == value2)
and
if(value == value1 && value == value2)
```
also this would help to not over use the command like
```
if(searcharray(.@array,"value") ||{&&} searcharray(.@array,"value2")
```
maybe this would be better

**Comment by secretdataz** [comment]
Since we don't know the precondition of the data we are going to search, the best we could do is O(n) (linear search) plus the cost of string comparison if the array is a string array. It could be more efficient than implementing linear search in the script tho. I'll consider implementing it in my free time.
(or you can open a PR, linear search is easy)

**Comment by Lemongrass3110** [comment]
Implemented in 151c847
